### DISCUSSION TITLE
IO cell library VDD mismatch

### DISCUSSION THREAD

**Discussion by bkoppelmann** [ucb-bar/hammer#724] [Support (Q&A)] [answered]
Hi,

My stdcells are using 1.2V, so I was using `vlsi.inputs.supplies.VDD: "1.2 V"`. However, my IO cells are using 3.3V. If I annotate the IO cell libs with
```
"supplies": {
    "VDD": "3.3 V",
    "GND": "0 V"
} 
```
then Hammer will not find those libs as I defined VDD to be 1.2 V. My workaround has been to just annotate the cell libs with the voltages I need. This feels like a hack to me. Is there a better way to do this?

Cheers,
Bastian

**Comment by harrisonliew** [comment, marked as answer]
Bastian,

Unfortunately, this is what you have to do for now. One could argue that for IO cells, what matters is the inward facing voltage, not the signaling voltage, which would still be 1.2V.

Anyways, we are slowly working on enabling true multi-power domain flows, in which you could specify multiple power domains and therefore be able to get all the relevant cell libraries.

**Comment by bkoppelmann** [comment]
Hi Harrison,

> One could argue that for IO cells, what matters is the inward facing voltage, not the signaling voltage, which would still be 1.2V.

Had the same thought right after I created the issue :). The workaround works for me. I just wanted to make sure I didn't overlook something. 

Thanks,
Bastian
